src/main/scala/com/wavesplatform/history/HistoryWriterImpl.scala

```diff
@@ -0,0 +1,82 @@
+package com.wavesplatform.history
+
+import com.google.common.cache.{CacheBuilder, CacheLoader}
+import com.google.common.util.concurrent.UncheckedExecutionException
+import scorex.account.Account
+import scorex.block.Block
+import scorex.block.Block.BlockId
+import scorex.transaction.History.BlockchainScore
+import scorex.transaction.ValidationError.CustomError
+import scorex.transaction.{History, HistoryWriter, ValidationError}
+import scorex.utils.ScorexLogging
+
+class HistoryWriterImpl(storage: HistoryStorage) extends History with HistoryWriter with ScorexLogging {
```

Comment:
`Either` for `appendBlock` and `Option` for `blockAt` are natural as to the semantics of what they are doing. Makeing `blockAt` `Either` doesn't make any sense.
Regarding exceptions, these are merely results of using cacheloader(not refactored at this point).
They do not get exposed so the usage is safe.

Anyway if you have an idea how to easily drop exceptions here(the problem is with sig `def load(height: Integer): Block`, please advice. We could return `null` in cacheloader on a miss, and then make `Option(..)` at `blockAt`, but that doesn't really change a lot.